lib/eco/ip_plugins/pythonindentation.py

```diff
@@ -13,12 +13,68 @@ def incparse_init(self):
         self.comment_tokens = []
         self.indent_stack = None
 
+    def right_sibling(self, node):
+        """Checks if a new indentation token was added or removed during the
+        current parse and returns the correct next node according to the current
+        parse tree instead of the previous version of the tree."""
+        if node in self.lookahead:
+            return self.lookahead[node]
+        else:
+            n = node.right_sibling(self.incparser.prev_version)
+            if n in self.deleted:
+                return self.right_sibling(n)
+            return n
+
+    def wasdeleted(self, node):
+        return node in self.deleted
+
+    def accept(self, bos, eos):
+        if self.wasdeleted(bos.next_term):
+            bos.next_term = bos.next_term.next_term
+        if self.wasdeleted(eos.prev_term):
+            eos.prev_term = eos.prev_term.prev_term
+
+    def reduce_process_child(self, c):
+        if isinstance(c.symbol, IndentationTerminal):
+            # this node is now part of the parse tree, thus properly update
+            # terminal links
+            c.prev_term.next_term = c
+            # We need to mark those nodes as changed so we can revert the
+            # changes when en error occurs. Unfortunately this slows down
+            # parsing because the parser now has to look down that tree even
+            # though the node has no parsing related changes
+            c.prev_term.mark_changed()
+            c.next_term.prev_term = c
+            c.next_term.mark_changed()
+        while self.wasdeleted(c.next_term):
+            c.next_term = c.next_term.next_term
+        while self.wasdeleted(c.prev_term):
+            c.prev_term = c.prev_term.prev_term
+
+    def pseudo_insert(self, oldnode, newnode):
+        #print("pseudo insert", oldnode.symbol.name, newnode.symbol.name)
+        self.lookahead[oldnode] = newnode
+        newnode.left = oldnode
+        newnode.right = oldnode.right
+        try:
+            print(newnode, "parent", newnode.parent.symbol.name, "changed to ", oldnode.parent.symbol.name)
+        except  AttributeError:
```

Comment:
How could I have overlooked this? I must have not been myself. Stupid mistake, Lukas, stupid!